fedimint-core/src/api.rs

```diff
@@ -613,6 +617,51 @@ impl WsClientConnectInfo {
     }
 }
 
+impl FromStr for WsClientConnectInfo {
+    type Err = anyhow::Error;
+
+    fn from_str(encoded: &str) -> Result<Self, Self::Err> {
+        let (hrp, data, variant) = bech32::decode(encoded)?;
+
+        ensure!(hrp == "fed", "Invalid HRP in bech32 encoding");
+        ensure!(variant == Bech32m, "Expected Bech32m encoding");
+
+        let bytes: Vec<u8> = Vec::<u8>::from_base32(&data)?;
+        let id_bytes: [u8; PK_SIZE] = bytes[..PK_SIZE].try_into()?;
+        let mut urls = vec![];
+        let mut index = PK_SIZE;
+
+        while index < bytes.len() {
+            let len = u16::from_be_bytes(bytes[index..index + 2].try_into()?) as usize;
+            index += 2;
+            let url = std::str::from_utf8(&bytes[index..index + len])?;
```

Comment:
This code crashes with malicious input (len greater than total byte len). We should just use a cursor instead of incrementing `index` and then use it as a `Read` object. I think that would also give us helpers for integer decoding. That should automatically give us sane errors when trying to read too much.